File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint_js/tp_sa1_dict_nonconstant_index_overapprox.js
// SA1 sound over-approximation: non-constant key at the WRITE site taints
// the whole container -- a later read of a DIFFERENT literal key still
// flags (never mute an unresolved-index write).
const m = {};

function handler(req, key) {
    m[key] = req.query.name;
    db.query(m["unrelated"]);
}

File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint/tp_sa1_dict_nonconstant_index_overapprox.py
"""SA1 sound over-approximation: the index is NON-constant (`key`, a
runtime variable) at the WRITE site, so the whole container must stay
tainted -- reading a different literal key afterwards still flags. Never
mute: an unresolved index is never treated as "safe"."""


def handler(key):
    m = {}
    m[key] = request.args["q"]
    cursor.execute(m["unrelated"])
